pyterrier/text.py

```diff
@@ -414,11 +432,15 @@ def transform(self, topics_and_res):
         return self.applyPassaging_no_qid(topics_and_res)
 
     def applyPassaging_no_qid(self, df):
-        p = re.compile(r"\s+")
+        if self.tokenizer is None:
+            p = re.compile(r"\s+")
         rows=[]
         for row in df.itertuples():
             row = row._asdict()
-            toks = p.split(row[self.text_attr])
+            if self.tokenizer is None:
+                toks = p.split(row[self.text_attr])
+            else:
+                toks = self.tokenizer.tokenize(row[self.text_attr])
```

Comment:
why note assign a local tokenizer variable:

```python
tokenizer = p.split if self.tokenizer is None else self.tokenizer.tokenize 

for bla:
  toks = tokenizer(row[self.text_attr])
```

This take an if statement out of the critical loop

Reply:
i've fixed this and added a unit-test as well